orchagent/routeorch.cpp

```diff
@@ -103,6 +103,83 @@ RouteOrch::RouteOrch(DBConnector *db, string tableName, NeighOrch *neighOrch) :
     m_syncdRoutes[v6_default_ip_prefix] = IpAddresses();
 
     SWSS_LOG_NOTICE("Create IPv6 default route with packet action drop");
+
+    /* All the interfaces have the same MAC address and hence the same
+     * auto-generated link-local ipv6 address with eui64 interface-id.
+     * Hence add a single /128 route entry for the link-local interface
+     * address pointing to the CPU port.
+     */
+    addLinkLocalRouteToMe(gVirtualRouterId);
+
+    /* TODO: Add the link-local /128 route to cpu in every VRF created from
+     * vrforch::addOperation. */
+}
+
+std::string RouteOrch::getLinkLocalEui64Addr(void)
+{
+    SWSS_LOG_ENTER();
+
+    string        ipPrefix;
+    const uint8_t *gmac = gMacAddress.getMac();
+
+    uint8_t        eui64_interface_id[EUI64_INTF_ID_LEN];
+    char           ipv6_ll_addr[IP6_ADDR_STR_LEN] = {0};
+  
+    eui64_interface_id[0] = gmac[0] ^ 0x02;
+    eui64_interface_id[1] = gmac[1];
+    eui64_interface_id[2] = gmac[2];
+    eui64_interface_id[3] = 0xff;
+    eui64_interface_id[4] = 0xfe;
+    eui64_interface_id[5] = gmac[3];
+    eui64_interface_id[6] = gmac[4];
+    eui64_interface_id[7] = gmac[5];
+
+    snprintf(ipv6_ll_addr, IP6_ADDR_STR_LEN, "fe80::%02x%02x:%02x%02x:%02x%02x:%02x%02x",
+             eui64_interface_id[0], eui64_interface_id[1], eui64_interface_id[2],
+             eui64_interface_id[3], eui64_interface_id[4], eui64_interface_id[5],
+             eui64_interface_id[6], eui64_interface_id[7]);
+
+    ipPrefix = string(ipv6_ll_addr);
+
+    return ipPrefix;
+}
+
+void RouteOrch::addLinkLocalRouteToMe(sai_object_id_t vrf_id)
```

Comment:
How about  passing `linklocal ipPrefix` also to this function so that if it is different for different interfaces/VRFs, the API would still work?

Reply:
Addressed the comment.